spec/models/capabilities/scopes/default_spec.rb

```diff
@@ -101,117 +90,181 @@
+            ]
+          end
+        end
+      end
     end
 
-    context 'with a global member without a permission' do
+    context 'with a global member without any permissions' do
       let(:members) { [global_member] }
 
-      it_behaves_like 'is empty'
+      include_examples 'is empty'
     end
 
-    context 'with a non member role without a permission' do
+    context 'with a non member role without any permissions' do
       let(:members) { [non_member_role] }
 
-      it_behaves_like 'is empty'
+      include_examples 'is empty'
+
+      context 'with the project being public and having a module activated with a public permission' do
+        before do
+          project.update(public: true)
+          project.enabled_module_names = ['activity']
+        end
+
+        include_examples 'consists of contract actions', with: 'the actions of the public permission' do
+          let(:expected) do
+            [
+              ['activities/read', user.id, project.id]
+            ]
+          end
+        end
+      end
     end
 
-    context 'with a global member with an action permission' do
+    context 'with a global member with a global permission' do
       let(:global_permissions) { %i[manage_user] }
       let(:members) { [global_member] }
 
-      it_behaves_like 'consists of contract actions' do
+      include_examples 'consists of contract actions', with: 'the actions of the global permission' do
         let(:expected) do
           [['users/create', user.id, nil],
            ['users/read', user.id, nil],
            ['users/update', user.id, nil]]
         end
       end
+
+      context 'with the user being locked' do
+        before do
+          user.locked!
+        end
+
+        include_examples 'is empty'
+      end
     end
 
-    context 'with a member with an action permission' do
+    context 'with a member with a project permission' do
       let(:permissions) { %i[manage_members] }
       let(:members) { [member] }
 
-      it_behaves_like 'consists of contract actions' do
+      include_examples 'consists of contract actions', with: 'the actions of the project permission' do
         let(:expected) do
           [['memberships/create', user.id, project.id],
            ['memberships/destroy', user.id, project.id],
            ['memberships/update', user.id, project.id]]
         end
       end
-    end
-
-    context 'with a global member with an action permission and the user being locked' do
-      let(:permissions) { %i[manage_user] }
-      let(:members) { [global_member] }
-      let(:user_status) { Principal.statuses[:locked] }
 
-      it_behaves_like 'is empty'
-    end
-
-    context 'with a member with an action permission and the user being locked' do
-      let(:permissions) { %i[manage_members] }
-      let(:members) { [member] }
-      let(:user_status) { Principal.statuses[:locked] }
+      context 'with the user being locked' do
+        before do
+          user.locked!
```

Comment:
The `locked!` method seems to be deprecated.
https://github.com/opf/openproject/blob/07927d1ce62c7125e3898a7ca293767cc93a83f2/app/models/user.rb#L316

Reply:
`lock!` is deprecated, and `locked!` should be used intead.

`deprecated_alias` signature is `deprecated_alias(old_method, new_method)`.

I suggest changing it to `deprecated_alias(dont_use:, use:)`, so that using it would look like
```ruby
  deprecated_alias dont_use: :lock!, use: :locked!
```

WDYT?